### PR TITLE
Electron

### PR DESCRIPTION
**Pull request by galec**
https://trello.com/c/IF2zhbuA/71-implement-electron

To run start npm run server and the do npm run eStart

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    // Specify entry point
+    win.loadURL('http://localhost:3000');
+
+
+    // build menu from template
+    /*  const mainMenu = Menu.buildFromTemplate(mainMenuTemplate);
+      Menu.setApplicationMenu(mainMenu); */
+
+    // Show dev tools
+    win.webContents.openDevTools();
```
Comment (cyrillegin) [inline main.js]: This is useful when debugging, however, our client won't be debugging and therefore doesn't need this window open. What would actually be really nice is if you could make a variable at the top or something called DEBUG, its default will be false but while we're developing, we can switch it to true to better help us debug.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+app.on('ready', () => {
+    createWindow();
+});
+
+app.on('window-all-closed', () => {
+    if (process.platform !== 'darwin') {
+        app.quit();
+    }
```
Comment (cyrillegin) [inline main.js]: I'm confused, if the platform === darwin, never quit?

File: package.json [hunk 3]
```diff
@@ -25,17 +26,19 @@
...
         "@uirouter/angularjs": "1.0.15",
-        "angular": "1.6.9",
+        "angular": "^1.6.9",
         "angular-animate": "1.6.9",
         "angular-aria": "1.6.9",
         "angular-route": "1.6.9",
         "d3": "4.13.0",
         "font-awesome": "4.7.0",
         "jquery": "3.3.1",
+        "node": "^9.9.0",
```
Comment (cyrillegin) [inline package.json]: This is unnecessary, near the top of the file, we list "node": "8.6.2" under engines. We don't actually want to package node into our node_modules do we? I could be wrong about electron needing node like this but it kind of feels weird.

File: package.json [hunk 4]
```diff
@@ -25,17 +26,19 @@
...
         "eslint": "./node_modules/.bin/eslint src",
         "tests": "mocha --require babel-register tests/test.js",
-        "cover": "node_modules/.bin/babel-node node_modules/.bin/babel-istanbul cover node_modules/.bin/_mocha -- tests/test.js"
+        "cover": "node_modules/.bin/babel-node node_modules/.bin/babel-istanbul cover node_modules/.bin/_mocha -- tests/test.js",
+        "eStart": "electron ."
     },
     "dependencies": {
         "@uirouter/angularjs": "1.0.15",
-        "angular": "1.6.9",
+        "angular": "^1.6.9",
```
Comment (cyrillegin) [inline package.json]: Please remove the carrot in front of the 1, we want to enforce exactly what version of each package we have.

File: package.json [hunk 5]
```diff
@@ -25,17 +26,19 @@
         "htmllint": "./node_modules/.bin/htmlhint --config .htmlhintrc.json src",
         "eslint": "./node_modules/.bin/eslint src",
         "tests": "mocha --require babel-register tests/test.js",
-        "cover": "node_modules/.bin/babel-node node_modules/.bin/babel-istanbul cover node_modules/.bin/_mocha -- tests/test.js"
+        "cover": "node_modules/.bin/babel-node node_modules/.bin/babel-istanbul cover node_modules/.bin/_mocha -- tests/test.js",
+        "eStart": "electron ."
```
Comment (cyrillegin) [inline package.json]: I could be wrong but isn't the convention to just have `"electron": "electron ."`? That way when you build, you run `npm run electron`.

File: main.js (hunk 2 above)
Comment (galec) [reply main.js]: on a mac it is common for application to stay active until the user input command q or at least that is what I understand from what I read

File: package.json (hunk 5 above)
Comment (galec) [reply package.json]: It is I just wanted something shorter I will change it.

**Comment by cyrillegin** [comment]
Looks good, nice job!

**Pull request merged**